src/test/java/org/junitpioneer/testkit/PioneerTestKit.java

```diff
@@ -38,12 +41,25 @@ public static ExecutionResults executeTestMethod(Class<?> testClass, String test
 	 *
 	 * @param testClass Name of the test class
 	 * @param testMethodName Name of the test method (of the given class)
-	 * @param methodParameterTypes Full qualified types names of the parameters (e.g. "java.nio.file.Path")
+	 * @param methodParameterTypes Class type(s) of the parameter(s)
 	 * @return The execution results
+	 * @throws IllegalArgumentException when methodParameterTypes is null
+	 * 			This method only checks parameters which are not part of the underlying
+	 * 			Jupiter TestKit. The Jupiter TestKit may throw other exceptions!
 	 */
 	public static ExecutionResults executeTestMethodWithParameterTypes(Class<?> testClass, String testMethodName,
-			String methodParameterTypes) {
-		return new ExecutionResults(testClass, testMethodName, methodParameterTypes);
+			Class<?>... methodParameterTypes) {
+
+		// throw IllegalArgumentException for a `null` array instead of NPE
+		// (hence no use of Objects::requireNonNull)`
+		if (methodParameterTypes == null) {
+			throw new IllegalArgumentException("methodParameterTypes must not be null");
+		}
```

Comment:
I removed the politics from the comment. My 2ct: `IllegalArgumentException` is the better exception, but nobody does it, so... :shrug: ... use `Optional`, I guess. :grin: